src/pydicom/pixels/decoders/gdcm.py

```diff
@@ -86,11 +87,23 @@ def _decode_frame(src: bytes, runner: DecodeRunner) -> bytes:
     ts_type = gdcm.TransferSyntax.GetTSType(str.__str__(tsyntax))
     img.SetTransferSyntax(gdcm.TransferSyntax(ts_type))
 
+    if tsyntax in uid.JPEGLSTransferSyntaxes:
+        bits_stored = runner.get_option("jls_precision")
+        if bits_stored in (6, 7):
+            raise ValueError(
+                "Unable to correctly decode JPEG-LS pixel data with a sample "
+                "precision of 6 or 7 bits"
```

Comment:
The JPEG-LS encoder may optimise 8-bit input down to the minimum required precision to improve compression ratio, but it's not like JPEG-LS is a common format. I only found it because I was writing tests as part of the JPEG-LS encoding PR and GDCM failed them. It's a very strange bug.